include/cuco/detail/trie/dynamic_bitset/dynamic_bitset.cuh

```diff
@@ -0,0 +1,359 @@
+  {
+    return (static_cast<uint64_t>(abs_hi_) << 8) | abs_lo_;
+  }
+
+  /**
+   * @brief Sets base rank of current 256-bit interval
+   *
+   * @param abs Base rank
+   */
+  __host__ __device__ constexpr void set_abs(uint64_t abs) noexcept
+  {
+    abs_hi_ = static_cast<uint32_t>(abs >> 8);
+    abs_lo_ = static_cast<uint8_t>(abs);
+  }
+};
+
+/**
+ * @brief Bitvector class with rank and select index structures
+ *
+ * In addition to standard bitvector get/set operations, this class provides
+ * rank and select operation API. It maintains index structures to make both these
+ * new operations close to constant time.
+ * Bitvector construction happens on host, after which the structures are moved to device.
+ * All subsequent read-only operations access device structures only.
+ *
+ * @tparam Allocator Type of allocator used for device storage
+ */
+// TODO: have to use device_malloc_allocator for now otherwise the container cannot grow
+template <class Allocator = thrust::device_malloc_allocator<std::byte>>
+class dynamic_bitset {
+ public:
+  using size_type = std::size_t;  ///< size type to specify bit index
+  using slot_type = uint64_t;     ///< Slot type
+  /// Type of the allocator to (de)allocate words
+  using allocator_type = typename std::allocator_traits<Allocator>::rebind_alloc<slot_type>;
+
+  static constexpr size_type words_per_block = 4;  ///< Tradeoff between space efficiency and perf.
+  static constexpr size_type bits_per_word   = sizeof(slot_type) * CHAR_BIT;     ///< Bits in a word
+  static constexpr size_type bits_per_block  = words_per_block * bits_per_word;  ///< Trivial
+
+  /**
+   * @brief Constructs an empty bitvector
+   *
+   * @param allocator Allocator used for allocating device storage
+   */
+  constexpr dynamic_bitset(Allocator const& allocator = Allocator{});
+
+  /**
+   * @brief adds a new bit at the end
+   *
+   * Grows internal storage if needed
+   *
+   * @param bit Boolean value of new bit to be added
+   */
+  constexpr void append(bool bit) noexcept;
+
+  /**
+   * @brief Modifies a single bit
+   *
+   * @param index position of bit to be modified
+   * @param bit new value of bit
+   */
+  constexpr void set(size_type index, bool bit) noexcept;
+
+  /**
+   * @brief Sets last bit to specified value
+   *
+   * @param bit new value of last bit
+   */
+  constexpr void set_last(bool bit) noexcept;
+
+  /**
+   * @brief Builds indexes for rank and select
+   */
+  constexpr void build() noexcept;
+
+  /**
+   * @brief Bulk get operation
+   *
+   * @tparam KeyIt Device-accessible iterator to keys
+   * @tparam OutputIt Device-accessible iterator to outputs
+   *
+   * @param keys_begin Begin iterator to keys list whose values are queried
+   * @param keys_end End iterator to keys list
+   * @param outputs_begin Begin iterator to outputs of get operation
+   * @param stream Stream to execute get kernel
+   */
+  template <typename KeyIt, typename OutputIt>
+  constexpr void get(KeyIt keys_begin,
```

Comment:
```suggestion
  constexpr void test(KeyIt keys_begin,
```
`test` instead of `get` to align with STL/boost.

Reply:
f96468df7b03a5ad8dd40eba9463f74fceb09142